### PR TITLE
Move the log into the database

### PR DESCRIPTION
**Pull request by micheles**
See https://bugs.launchpad.net/oq-engine/+bug/1430206.
The tests are running here: https://ci.openquake.org/job/zdevel_oq-engine/1089

### CODE REVIEW

File: openquake/engine/utils/tasks.py [hunk 1]
```diff
@@ -157,10 +157,14 @@ def wrapped(*args):
         # the revoke functionality can work
         EnginePerformanceMonitor.store_task_id(job_id, tsk)
 
+        # if there is no db handler already, add one
+        if not any(isinstance(handler, logs.LogDatabaseHandler)
+                   for handler in logs.LOG.handlers):
+            logs.LOG.addHandler(logs.LogDatabaseHandler(job))
+            logs.set_level(job.log_level)
+
```
Comment (nastasi-oq) [inline openquake/engine/utils/tasks.py]: What happens when more than one job concur on the same worker ?
Add a test to check this behaviour could be a good choice.

File: openquake/engine/utils/tasks.py [hunk 2]
```diff
@@ -158,9 +158,8 @@ def wrapped(*args):
         EnginePerformanceMonitor.store_task_id(job_id, tsk)
 
         with EnginePerformanceMonitor(
-                'total ' + task_func.__name__, job_id, tsk, flush=True):
-            # tasks write on the celery log file
-            logs.set_level(job.log_level)
+                'total ' + task_func.__name__, job_id, tsk, flush=True
+        ), logs.handle(job):
             try:
```
Comment (nastasi-oq) [inline openquake/engine/utils/tasks.py]: Use a couple of () to group the 2 with statements seems to me more correct.

**Comment by nastasi-oq** [comment]
LGTM

File: openquake/engine/utils/tasks.py (hunk 1 above)
Comment (micheles) [inline openquake/engine/utils/tasks.py]: I have change the approach to use a with statement which is cleaner and safer

File: openquake/engine/utils/tasks.py (hunk 2 above)
Comment (micheles) [inline openquake/engine/utils/tasks.py]: Done

**Pull request merged**